src/App.tsx

```diff
@@ -1,14 +1,47 @@
 /* eslint-disable max-len */
-import React from 'react';
+import { FC, useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
+export enum Status {
+  all = 'all',
+  active = 'active',
+  completed = 'completed',
+}
+
+export const App: FC = () => {
+  const [status, setStatus] = useState<Status>(Status.all);
+  const [query, setQuery] = useState<string>('');
+  const [todosFromServer, setTodosFromServer] = useState<Todo[]>([]);
+  const [isOpenedTodo, setIsOpenedTodo] = useState<Todo | null>(null);
+  let displayedTodos = [...todosFromServer];
+
+  if (status === Status.active) {
+    displayedTodos = todosFromServer.filter(todo => !todo.completed);
+  }
+
+  if (status === Status.completed) {
+    displayedTodos = todosFromServer.filter(todo => todo.completed);
+  }
+
+  if (query) {
+    displayedTodos = displayedTodos.filter(todo =>
+      todo.title.toLowerCase().includes(query.toLowerCase()),
+    );
+  }
```

Comment:
what if we introduce a separate state for displayedTodos + util function?

It can look like this:

```suggestion
  const [displayedTodos, setDisplayedTodos] = useState(
    getDisplayedTodos(status, query)
  );
```

And the function will handle the stuff inside instead of keeping this low level logic in the component